src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -613,22 +609,47 @@ bool LibraryCallKit::inline_vector_shuffle_iota() {
   int num_elem = vlen->get_con();
   BasicType elem_bt = T_BYTE;
 
-  if (!arch_supports_vector(VectorNode::replicate_opcode(elem_bt), num_elem, elem_bt, VecMaskNotUsed)) {
+  bool effective_indices_in_range = false;
+  if (start_val->is_con() && step_val->is_con()) {
+    int effective_min_index = start_val->get_con();
+    int effective_max_index = start_val->get_con() + step_val->get_con() * (num_elem - 1);
+    effective_indices_in_range = effective_min_index >= -128 && effective_max_index <= 127;
+  }
```

Comment:
May I ask why we need to fall-back to java implementation if the indices are in-effective for constant vals?  While if the `start_val` and `step_val` are not all constants, for in-effective-indices, it subs to the lanecount? Is there any difference for constant and variable inputs?

Reply:
An alternative is moving this effective indice checking for constant values in java level. C2 compiler may optimize out it for most cases?

Reply:
> An alternative is moving this effective indice checking for constant values in java level. C2 compiler may optimize out it for most cases?

I wanted to just fix incorrectness issue with this PR given that we already have another JBS (JDK-8310691) for shuffle related overhaul in progress.

> May I ask why we need to fall-back to java implementation if the indices are in-effective for constant vals? While if the `start_val` and `step_val` are not all constants, for in-effective-indices, it subs to the lanecount? Is there any difference for constant and variable inputs?

Inline expander operate over byte vectors currently, this check ensures that we do not overflow the byte value range while computing effective index for unwrapped argument, otherwise subsequent comparison mask may be incorrectly.